test_scripts/Protocol/NACKReason/001_Second_Protected_Video_Service_NACK.lua

```diff
@@ -0,0 +1,108 @@
+---------------------------------------------------------------------------------------------------
+--  Precondition:
+--  1) Initialize the client side certifcate file for SDL
+--  2) Start SDL, HMI, connect Mobile device
+--  3) Register App_1 on SDL.
+--
+--  Steps:
+--  1) Send StartService Request(with protocol version 5.3.0) to switch the RPC Service to Protected mode
+--  SDL Does:
+--    a) Send a StartService ACK message
+--  2) Send a StartService Request to start a protected VIDEO service
+--  SDL Does:
+--    a) Send a StartService ACK message
+--  3) Send another StartService Request to start a second protected VIDEO service
+--  SDL Does:
+--    a) Send a StartService NACK message with a reason parameter in the bson payload
+--  4) Unregister and re-register App_1
+--  5) Send StartService Request(with protocol version 5.2.0) to switch the RPC Service to Protected mode
+--  SDL Does:
+--    a) Send a StartService ACK message
+--  6) Send a StartService Request to start a protected VIDEO service
+--  SDL Does:
+--    a) Send a StartService ACK message
+--  7) Send another StartService Request to start a second protected VIDEO service
+--  SDL Does:
+--    a) Send a StartService NACK message with an empty bson payload(no reason param)
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local utils = require("user_modules/utils")
+if not utils.isFileExist("lib/bson4lua.so") then
+  runner.skipTest("'bson4lua' library is not available in ATF")
```

Comment:
should this test skip or fail if there is no bson lib? I just wonder with the atf runner if we'd ever notice such an error if we just skip the test.

Reply:
I see the reasoning behind failing the tests if the bson_lib isn't there. I guess one possible scenario for the skipping the tests might be if we were purposely testing without the bson_lib, in which case skipping the tests would make more sense than marking them as failed tests. I'm fine with either approach, let me know what you think.